src/fdb5/database/Index.h

```diff
@@ -108,7 +108,9 @@ class IndexBase : public eckit::Counted {
     void decodeCurrent(eckit::Stream& s, const int version);
     void decodeLegacy(eckit::Stream& s, const int version);
 
-    virtual void add(const Key &key, const Field &field) = 0;
+    virtual void add(const Key& key, const Field &field) = 0;
+
+    const TypesRegistry& registry() const;
```

Comment:
`protected registry_` but a `private const TypesRegistry& registry() const;` tells me this needs more attention